src/quo2/components/community/token_gating.cljs

```diff
@@ -0,0 +1,166 @@
+                       :margin-horizontal -20
+                       :margin-top 12
+                       :margin-bottom 8}
+             :enter-button {:margin-top 16}
+             :info-text {:margin-top 12
+                         :text-align :center
+                         :padding-horizontal 20}})
+
+(defn is-token-requirement-met? [token-requirement-list]
+  (every?
+   (fn [token]
+     (get token :is-sufficient?))
+   token-requirement-list))
+
+(defn token-requirement-list [props]
+  (let [{:keys [gate]} props
+        [gate-type token-requirement-list] gate
+        is-sufficient? (is-token-requirement-met? token-requirement-list)
+        not-text (if-not is-sufficient? " not" "")
+        message (if
+                 (= gate-type :join)
+                  (i18n/label
+                   :t/community-join-requirements {:not not-text})
+                  (if (= gate-type :read)
+                    (i18n/label
+                     :t/community-channel-read-requirements {:not not-text})
+                    (i18n/label
+                     :t/community-channel-write-requirements {:not not-text})))]
+    [rn/view
+     [rn/view {:style (get styles :token-requirement-list-spacing)}
+      [text/text {:weight :medium
+                  :number-of-lines 1} message]
+      [text/text {:size :paragraph-2
+                  :number-of-lines 1} "You must hodl:"]]
+
+     [rn/view {:style (merge (get styles :token-row) (merge (get styles :token-requirement-list-spacing)) (get styles :token-list))}
+      (map-indexed (fn [token-index token]
+                     ^{:key token}
+                     [:<>
+                      [rn/view {:style (when-not (= token-index 0) (get styles :token-tag-spacing))}
+                       [token-tag/token-tag {:token (:token token)
+                                             :value (:amount token)
+                                             :size :big
+                                             :is-required (:is-sufficient? token)
+                                             :is-purchasable (:is-purchasable token)
+                                             :token-img-src
+                                             (js/require
+                                              (str
+                                               "../resources/images/tokens/mainnet/"
+                                               (:token token)
+                                               ".png"))}]]])
+                   token-requirement-list)]]))
+
+(defn is-community-locked? [community]
+  (let [{gates :gates} community]
+    (when (vector? (:join gates)
+                   (not (is-token-requirement-met? (:join gates)))))))
+
+(defn is-channel-locked? [channel]
+  (let [{gates :gates} channel]
+    (if (vector? (:read gates))
+      (not (is-token-requirement-met? (:read gates)))
+      (when (vector? (:write gates))
+        (not (is-token-requirement-met? (:write gates)))))))
```

Comment:
we need to check every time if it is vector, while calling `is-token-requirement-met?` function, maybe better to move this check to function itself.

Reply:
so, this will become
```
(defn is-channel-locked? [channel]
  (let [{gates :gates} channel]
    (not (is-token-requirement-met? (or (:read gates) (:write gates)))
```

Reply:
Checking for `vector?` is necessary, because i want to check which "gates" are set. Moving this to the enclosing function would require us to add state, since the gates from the props can change

Reply:
if no `:read` gate is given, then this function will check if there is a `:write` gate...

Reply:
current implementation will return true for both `vector?` and `is-token-requirement-met?` if empty array is passed `[]`

You can do something like
```
(defn is-token-requirement-met? [token-requirement-list]
  (if (seq token-requirement-list)
    (every?
     (fn [token]
       (get token :is-sufficient?))
     token-requirement-list)
    false))
```
> if no :read gate is given, then this function will check if there is a :write gate...

And, for `is-channe-locked?`
```
(defn is-channel-locked? [channel]
  (let [{gates :gates} channel]
    (not (is-token-requirement-met? (or (:read gates) (:write gates)))   <------ This will make sure to pass whichever is available
```

Reply:
updated the function with `and`

Reply:
thanks! 👍